src/templates/partials/redis.hbs

```diff
@@ -14,8 +14,15 @@ tls-ca-cert-file /path/to/ca_certificates.crt
 tls-ca-cert-dir /path/to/ca_certificates
 
 {{#if output.usesDhe}}
+{{#if (minver "6.2.7" form.serverVersion)}}
+{{#unless (minver "3.0.0" form.opensslVersion)}}
 # {{output.dhCommand}} > /path/to/dhparam
 tls-dh-params-file /path/to/dhparam
+{{/unless}}
+{{else}}
+# {{output.dhCommand}} > /path/to/dhparam
+tls-dh-params-file /path/to/dhparam
+{{/if}}
```

Comment:
OK would you be willing to add an "AND" helper (e.g. https://github.com/helpers/handlebars-helpers/blob/master/lib/comparison.js …) to avoid this repetition all over the place, please?

Reply:
Seems doable.  Time allowing, I'll try to revisit this on Sun.

If this existing code in this PR works correctly, I would prefer this PR not be blocked for this kind of polishing.

Reply:
Turns out to not be trivial.
Examples in https://stackoverflow.com/questions/8853396/logical-operator-in-a-handlebars-js-if-conditional work on simple values but not on items which need to be evaluated.

Some possible (not tested) solutions include  https://github.com/jmurphyau/ember-truth-helpers and or `lvalue = Embers.Handlerbars.get(this, lvalue) || lvalue` or https://gist.github.com/servel333/21e1eedbd70db5a7cfff327526c72bc5

Possible solutions need to be evaluated for security XSS.

A better solution, IMHO, is to hoist logic which needs multiple conditions out of the templates and into javascript, and have the templates access the result.  I don't think adding more dependencies and bloating the generated page is the best solution.  Please create a separate github issue if you would like to evaluate options.  At the moment, the small bits of duplication in the templates is a workable solution for this PR.

Reply:
Thanks for taking a stab at these… that unfortunately sounds awfully familiar from when we were adding some heavy hbs logic few years ago. I will consult what we ended up using and with how much success regarding the overall experience and flexibility…

(I don't follow the Ember note as we probably won't be adding a framework to facilitate any rendering when we have the current webpack stack with loaders etc. serving that right from the single bundled app…)

EDIT: This has solved itself over time as the repetition for defaulting to auto groups for various versions is not in the scope of this PR anymore.

EDIT2: I had quickly adapted some of our helpers ([e.g.](https://github.com/janbrasna/ssl-config-generator/compare/9e96f08...7d7dd2), +[demo](https://fix-awselb-dhe--mozsslconf-dev.netlify.app/#server=awselb)) and had no issues besides some inverted logic from what I was used to. Ugly, only way without moving away from HBS first, but best you can do and it seems to work for what we need. (This example should not ever be necessary again, as it's an ugly hack for a reason; normally you need just a nice `(and … … …)` plus its inversion. It only needs some more work to turn the expression `{{when}}` returning the outputs, needing `{{#if (when …)}}`, to block proper, allowing `{{#when …}}{{/when}}` blocks directly.)